### DISCUSSION TITLE
How to check if room exists/valid

### DISCUSSION THREAD

**Discussion by Mathiaszero** [miguelgrinberg/python-socketio#1412] [Q&A]
I made function to get rooms from default namespace when 2 clients are connected using `items = sio.manager.rooms['/'].items()`, however it shows this:
![image](https://github.com/user-attachments/assets/3eeeb71a-b626-4645-bc04-d26f08500b23)

It shows a mix of a none type, both clients (Jz and iB), and the room (sub_...) as the keys, which will be an issue for me. Is there something like `RoomExists(room)` that can return true or false if it is valid room or not?

**Comment by miguelgrinberg** [comment]
You can do a lot of things if you access the internal data structures that this package maintains, but this isn't supported and is likely to break in future releases as this package evolves. These structures just exist to support the public interfaces of the package, which is what you should use instead.

The design for rooms is that they are implicitly created and destroyed as needed, if really doesn't make sense to create a room or to check if a room exists. If you need to know what rooms are available, then you should track these rooms in your own application.
